cognite/neat/graph/loaders/rdf_to_dms.py

```diff
@@ -86,11 +112,41 @@ def rdf2nodes_and_edges(
 
                     if isinstance(e, NeatException):
                         exceptions.append(e.to_error_dict())
+                    else:
+                        exceptions.append(
+                            ErrorDetails(
+                                input=class_instance_id,
+                                loc=tuple(["rdf2nodes_and_edges"]),
+                                msg=str(e),
+                                type=f"Exception of type {type(e).__name__} occurred  \
+                                when processing instance of {class_}",
+                            )
+                        )
                     continue
 
     return nodes, edges, exceptions
 
 
+def is_node_valid(node: NodeApply) -> tuple[bool, str]:
+    if node.external_id is None or node.external_id == "" or len(node.external_id) >= 255:
+        return False, f"external_id {node.external_id} is empty of too long"
+    return True, ""
+
+
+def is_edge_valid(edge: EdgeApply) -> tuple[bool, str]:
+    if edge.external_id is None or edge.external_id == "" or len(edge.external_id) >= 255:
+        return False, f"external_id {edge.external_id} is empty of too long"
+    if (
+        edge.start_node.external_id is None
+        or edge.start_node.external_id == ""
+        or len(edge.start_node.external_id) >= 255
+    ):
+        return False, f"start_node.external_id {edge.start_node.external_id} is empty of too long"
+    if edge.end_node.external_id is None or edge.end_node.external_id == "" or len(edge.end_node.external_id) >= 255:
+        return False, f"end_node.external_id {edge.end_node.external_id} is empty of too long"
+    return True, ""
```

Comment:
Lets pull out `is_valid_external_id`
```suggestion
def is_node_valid(node: NodeApply) -> tuple[bool, str]:
    return is_valid_external_id(node.external_id)


def is_edge_valid(edge: EdgeApply) -> tuple[bool, str]:
    for external_id in [edge.external_id, edge.start_node.external_id, edge.end_node.external_id]:
        is_valid, reason = is_valid_external_id(external_id)
        if not is_valid:
            return False, reason
    return True, ""

def is_valid_external_id(external_id: str) -> tuple[bool, str]:
    if external_id is None or external_id == "" or len(external_id) >= 255:
        return False, f"external_id {external_id} is empty of too long"
    return True, ""
```

Less code to maintain and easier to read.